pkg/apis/policy/v1alpha1/register.go

```diff
@@ -41,6 +42,7 @@ func addKnownTypes(scheme *runtime.Scheme) {
 	scheme.AddKnownTypes(SchemeGroupVersion,
 		&PodDisruptionBudget{},
 		&PodDisruptionBudgetList{},
+		&v1.ListOptions{},
```

Comment:
We can do it in another issue - it's no less broken then (everything in an
API is versioned, even if we haven't explicitly created a second versioned
yet).  Said another way, unversioned is something we haven't correctly
versioned yet (a bug).

It's not really an API change - the schema accepted is a version, so it's
really just an internal error that each group version doesn't explicitly
say what schema of export options they support.

On Tue, Jun 21, 2016 at 4:16 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/apis/policy/v1alpha1/register.go
> <https://github.com/kubernetes/kubernetes/pull/27733#discussion_r67943081>
> :
>
> > @@ -41,6 +42,7 @@ func addKnownTypes(scheme *runtime.Scheme) {
> >  	scheme.AddKnownTypes(SchemeGroupVersion,
> >  		&PodDisruptionBudget{},
> >  		&PodDisruptionBudgetList{},
> > +		&v1.ListOptions{},
>
> ExportOptions is registered as unversioned type:
> https://github.com/kubernetes/kubernetes/blob/master/pkg/api/register.go#L108
> .
>
> We should register it as an ordinary type. @smarterclayton
> <https://github.com/smarterclayton> Are you fine if I do that in another
> PR? Moving ExportOptions from unversioned/ to a api/ is an API change.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27733/files/1c74f17935c2b2cd29db72eda4f7a271b9d3bf24#r67943081>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p_9PwWxieo6bTCjo3x8TS_UAqHvEks5qOEaCgaJpZM4I6N5r>
> .
>